channel/persistence/test/peerchans.go

```diff
@@ -73,10 +74,17 @@ func (pc peerChans) Delete(id channel.ID) {
 	}
 }
 
-func peerKey(a wire.Address) string { return string(a.Bytes()) }
+func peerKey(a wire.Address) string {
+	var key = new(bytes.Buffer)
+	err := io.Encode(key, a)
+	if err != nil {
+		panic("error encoding peer key: " + err.Error())
+	}
+	return key.String()
+}
 
 func peerFromKey(s string) wire.Address {
-	p, err := wire.DecodeAddress(strings.NewReader(s))
+	p, err := wire.DecodeAddress(bytes.NewBuffer([]byte(s)))
```

Comment:
Why not use binary marshaller?

Reply:
We could do that, but it would require a lot of changes within the `keyvalue` package. Because, in  current implementation, encoding/decoding is widely used for storing/retrieving data.

However, thanks for pointing out, because I just noticed `DecodeAddress` is not replaced with `perunio.Decode` here. I have fixed this in 658c45bcb5d4ceb29ee39f913d4e1a9bfaaf3c62.